### PR TITLE
Updated error class and tests

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
As soon as #32 was merged I realized I could've added a better test. Now the test checks that the suggested option for `invalidOption` was `intOption`.

I have also updated the `Error` class slightly:
- added a `message` field to store the _unformatted_ message that's passed in
- instead of manually printing the message, pass it to the parent class for printing. That way, you can do things like
```
try:
    <some code>
except Exception as e:
    print(f"Something went wrong! The exception raised was {str(e)}")
```
or similar where you can access the _formatted_ message via `str(e)`, and unformatted message via `e.message`.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
Explain the steps needed to test the new code to verify that it does indeed address the issue and produce the expected behavior.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [x] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: baseclasses/utils.py [hunk 1]
```diff
@@ -73,5 +74,4 @@ def __init__(self, message):
                 msg += word + " "
                 i += len(word) + 1
         msg += " " * (78 - i) + "|\n" + "+" + "-" * 78 + "+" + "\n"
-        print(msg)
-        super().__init__()
+        super().__init__(msg)
```
Comment (sseraj) [inline baseclasses/utils.py:77]: Does this print the message at all? I tried it on ADflow and it doesn't work.

Comment (nwu63) [reply baseclasses/utils.py:77]: Huh really? It works fine with me. Try the following MWE?
```
from baseclasses.utils import Error
raise Error('Failed!')
```

Comment (sseraj) [reply baseclasses/utils.py:77]: That works. Maybe it has something to do with Fortran.

Comment (sseraj) [reply baseclasses/utils.py:77]: pyHyp and IDWarp both print the error messages, so it might just a problem with ADflow.

Comment (sseraj) [reply baseclasses/utils.py:77]: Okay, it seems like the problem was on my end with running a `unittest` script with `python`. This works for me on a regular ADflow run script or for a `unittest` script with `testflo`.

**Review by sseraj — APPROVED** [review]

**Review by marcomangano — APPROVED** [review]

**Pull request merged**